src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java

```diff
@@ -39,9 +39,14 @@ public String getToken() {
     // Called from other synchronized code. Do not call into other synchronized code or you'll
     // risk deadlock
     public synchronized void setDistinctId(String distinctId) {
-        mUnseenSurveys.clear();
-        mUnseenNotifications.clear();
-        mDistinctId = distinctId;
+        if (mDistinctId == distinctId) {
```

Comment:
if (mDistinctId != null && mDistinctId.equals(distinctId)) 
I think this should be preferred.